animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,145 @@
+                    return "execute_IndicesSingle"
+            else:
+                return "execute_Index"
+        elif self.mode == "ALL":
+            if self.useFloatList:
+                return "execute_AllList"
+            else:
+                return "execute_AllSingle"
+
+    def execute_Index(self, object, identifier, index, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        vertexGroup.add([index], weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesSingle(self, object, identifier, indices, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_IndicesList(self, object, identifier, indices, weights):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        for i, index in enumerate(indices):
+            vertexGroup.add([index], weights[i], "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllSingle(self, object, identifier, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        indices = list(range(0, len(object.data.vertices)))
+        vertexGroup.add(indices, weight, "REPLACE")
+        object.data.update()    
+        return object
+
+    def execute_AllList(self, object, identifier, weights):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+
+        totalVertices = len(object.data.vertices)    
+        for i, weight in enumerate(weights):
+            if i >= totalVertices: return 
+            vertexGroup.add([i], weight, "REPLACE") 
+        object.data.update()
+        return object            
+     
+    def getVertexGroup(self, object, identifier):
+        try: return object.vertex_groups[identifier]
+        except: return object.vertex_groups.new()
```

Comment:
Why create a new group? If no group exists, just raise an exception.

Reply:
If there is no group then Set Vertex Weights node will create a new one otherwise we have to create it manually from the Object Vertex Tab. To avoid this manual step, I have added this option.

Reply:
But it is not intuitive and unexpected to the user. The node is supposed to **Set** Vertex Weights, not create vertex groups. If you think it is important to have a fast way to create vertex groups, then I think we should add an operator somewhere in the node, not add them implicitly.

Reply:
Okay, I'll remove this and raise an exception.